File: backend/src/main/java/com/inmobiliaria/mantenimiento/service/MantenimientoService.java
package com.inmobiliaria.mantenimiento.service;

import com.inmobiliaria.mantenimiento.domain.*;
import com.inmobiliaria.mantenimiento.dto.*;
import com.inmobiliaria.mantenimiento.repository.*;
import com.inmobiliaria.shared.multitenancy.TenantContext;
import jakarta.persistence.EntityNotFoundException;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
@Slf4j
public class MantenimientoService {

    private final ProveedorRepository proveedorRepository;
    private final OrdenMantenimientoRepository ordenRepository;
    private final SeguimientoOrdenRepository seguimientoRepository;

    // ==================== PROVEEDORES ====================

    @Transactional(readOnly = true)
    public List<ProveedorDTO> findAllProveedores() {
        Long empresaId = TenantContext.getCurrentTenant();
        return proveedorRepository.findByEmpresaIdOrderByNombreAsc(empresaId)
                .stream()
                .map(ProveedorDTO::fromEntity)
                .collect(Collectors.toList());
    }

    @Transactional(readOnly = true)
    public List<ProveedorDTO> findProveedoresActivos() {
        Long empresaId = TenantContext.getCurrentTenant();
        return proveedorRepository.findByEmpresaIdAndActivoTrueOrderByNombreAsc(empresaId)
                .stream()
                .map(ProveedorDTO::fromEntity)
                .collect(Collectors.toList());
    }

    @Transactional(readOnly = true)
    public ProveedorDTO findProveedorById(Long id) {
        Long empresaId = TenantContext.getCurrentTenant();
        Proveedor proveedor = proveedorRepository.findById(id)
                .filter(p -> p.getEmpresaId().equals(empresaId))
                .orElseThrow(() -> new EntityNotFoundException("Proveedor no encontrado"));
        return ProveedorDTO.fromEntity(proveedor);
    }

    @Transactional(readOnly = true)
    public List<ProveedorDTO> findProveedoresByCategoria(CategoriaMantenimiento categoria) {
        Long empresaId = TenantContext.getCurrentTenant();
        return proveedorRepository.findActivosByEmpresaIdAndCategoria(empresaId, categoria)
                .stream()
                .map(ProveedorDTO::fromEntity)
                .collect(Collectors.toList());
    }

    @Transactional
    public ProveedorDTO createProveedor(CreateProveedorRequest request) {
        Long empresaId = TenantContext.getCurrentTenant();

        Proveedor proveedor = Proveedor.builder()
                .empresaId(empresaId)
                .nombre(request.getNombre())
                .razonSocial(request.getRazonSocial())
                .rfc(request.getRfc())
                .telefonoPrincipal(request.getTelefonoPrincipal())
                .telefonoSecundario(request.getTelefonoSecundario())
                .email(request.getEmail())
                .direccion(request.getDireccion())
                .codigoPostal(request.getCodigoPostal())
                .ciudad(request.getCiudad())
                .estado(request.getEstado())
                .nombreContacto(request.getNombreContacto())
                .categorias(request.getCategorias())
                .notas(request.getNotas())
                .activo(true)
                .build();

        proveedor = proveedorRepository.save(proveedor);
        log.info("Proveedor creado: {}", proveedor.getId());
        return ProveedorDTO.fromEntity(proveedor);
    }

    @Transactional
    public ProveedorDTO updateProveedor(Long id, UpdateProveedorRequest request) {
        Long empresaId = TenantContext.getCurrentTenant();
        Proveedor proveedor = proveedorRepository.findById(id)
                .filter(p -> p.getEmpresaId().equals(empresaId))
                .orElseThrow(() -> new EntityNotFoundException("Proveedor no encontrado"));

        proveedor.setNombre(request.getNombre());
        proveedor.setRazonSocial(request.getRazonSocial());
        proveedor.setRfc(request.getRfc());
        proveedor.setTelefonoPrincipal(request.getTelefonoPrincipal());
        proveedor.setTelefonoSecundario(request.getTelefonoSecundario());
        proveedor.setEmail(request.getEmail());
        proveedor.setDireccion(request.getDireccion());
        proveedor.setCodigoPostal(request.getCodigoPostal());
        proveedor.setCiudad(request.getCiudad());
        proveedor.setEstado(request.getEstado());
        proveedor.setNombreContacto(request.getNombreContacto());
        if (request.getCategorias() != null) {
            proveedor.setCategorias(request.getCategorias());
        }
        proveedor.setNotas(request.getNotas());
        if (request.getActivo() != null) {
            proveedor.setActivo(request.getActivo());
        }

        proveedor = proveedorRepository.save(proveedor);
        log.info("Proveedor actualizado: {}", proveedor.getId());
        return ProveedorDTO.fromEntity(proveedor);
    }

    @Transactional
    public void deleteProveedor(Long id) {
        Long empresaId = TenantContext.getCurrentTenant();
        Proveedor proveedor = proveedorRepository.findById(id)
                .filter(p -> p.getEmpresaId().equals(empresaId))
                .orElseThrow(() -> new EntityNotFoundException("Proveedor no encontrado"));

        proveedorRepository.delete(proveedor);
        log.info("Proveedor eliminado: {}", id);
    }

    // ==================== ÓRDENES DE MANTENIMIENTO ====================

    @Transactional(readOnly = true)
    public List<OrdenMantenimientoDTO> findAllOrdenes() {
        Long empresaId = TenantContext.getCurrentTenant();
        return ordenRepository.findByEmpresaIdOrderByFechaCreacionDesc(empresaId)
                .stream()
                .map(OrdenMantenimientoDTO::fromEntity)
                .collect(Collectors.toList());
    }

    @Transactional(readOnly = true)
    public List<OrdenMantenimientoDTO> findOrdenesActivas() {
        Long empresaId = TenantContext.getCurrentTenant();
        return ordenRepository.findOrdenesActivas(empresaId)
                .stream()
                .map(OrdenMantenimientoDTO::fromEntity)
                .collect(Collectors.toList());
    }

    @Transactional(readOnly = true)
    public OrdenMantenimientoDTO findOrdenById(Long id) {
        Long empresaId = TenantContext.getCurrentTenant();
        OrdenMantenimiento orden = ordenRepository.findById(id)
                .filter(o -> o.getEmpresaId().equals(empresaId))
                .orElseThrow(() -> new EntityNotFoundException("Orden de mantenimiento no encontrada"));
        return OrdenMantenimientoDTO.fromEntity(orden);
    }

    @Transactional(readOnly = true)
    public List<OrdenMantenimientoDTO> findOrdenesByPropiedad(Long propiedadId) {
        Long empresaId = TenantContext.getCurrentTenant();
        return ordenRepository.findByEmpresaIdAndPropiedadIdOrderByFechaCreacionDesc(empresaId, propiedadId)
                .stream()
                .map(OrdenMantenimientoDTO::fromEntity)
                .collect(Collectors.toList());
    }

    @Transactional(readOnly = true)
    public List<OrdenMantenimientoDTO> findOrdenesByProveedor(Long proveedorId) {
        Long empresaId = TenantContext.getCurrentTenant();
        return ordenRepository.findByEmpresaIdAndProveedorIdOrderByFechaCreacionDesc(empresaId, proveedorId)
                .stream()
                .map(OrdenMantenimientoDTO::fromEntity)
                .collect(Collectors.toList());
    }

    @Transactional(readOnly = true)
    public List<OrdenMantenimientoDTO> findOrdenesByEstado(EstadoOrden estado) {
        Long empresaId = TenantContext.getCurrentTenant();
        return ordenRepository.findByEmpresaIdAndEstadoOrderByPrioridadDescFechaSolicitudAsc(empresaId, estado)
                .stream()
                .map(OrdenMantenimientoDTO::fromEntity)
                .collect(Collectors.toList());
    }

    @Transactional(readOnly = true)
    public List<OrdenMantenimientoDTO> findOrdenesProgramadas(LocalDate inicio, LocalDate fin) {
        Long empresaId = TenantContext.getCurrentTenant();
        return ordenRepository.findProgramadasEnRango(empresaId, inicio, fin)
                .stream()
                .map(OrdenMantenimientoDTO::fromEntity)
                .collect(Collectors.toList());
    }

    @Transactional
    public OrdenMantenimientoDTO createOrden(CreateOrdenRequest request) {
        Long empresaId = TenantContext.getCurrentTenant();

        OrdenMantenimiento orden = OrdenMantenimiento.builder()
                .empresaId(empresaId)
                .propiedadId(request.getPropiedadId())
                .proveedorId(request.getProveedorId())
                .solicitanteId(request.getSolicitanteId())
                .titulo(request.getTitulo())
                .descripcion(request.getDescripcion())
                .categoria(request.getCategoria())
                .prioridad(request.getPrioridad())
                .estado(EstadoOrden.PENDIENTE)
                .fechaSolicitud(LocalDate.now())
                .fechaProgramada(request.getFechaProgramada())
                .costoEstimado(request.getCostoEstimado())
                .notasTecnicas(request.getNotasTecnicas())
                .build();

        orden = ordenRepository.save(orden);

        // Registrar seguimiento inicial
        registrarSeguimiento(orden.getId(), null, EstadoOrden.PENDIENTE, "Orden creada");

        log.info("Orden de mantenimiento creada: {}", orden.getId());
        return OrdenMantenimientoDTO.fromEntity(orden);
    }

    @Transactional
    public OrdenMantenimientoDTO updateOrden(Long id, UpdateOrdenRequest request) {
        Long empresaId = TenantContext.getCurrentTenant();
        OrdenMantenimiento orden = ordenRepository.findById(id)
                .filter(o -> o.getEmpresaId().equals(empresaId))
                .orElseThrow(() -> new EntityNotFoundException("Orden de mantenimiento no encontrada"));

        EstadoOrden estadoAnterior = orden.getEstado();

        orden.setProveedorId(request.getProveedorId());
        orden.setSolicitanteId(request.getSolicitanteId());
        orden.setTitulo(request.getTitulo());
        orden.setDescripcion(request.getDescripcion());
        orden.setCategoria(request.getCategoria());
        orden.setPrioridad(request.getPrioridad());
        if (request.getEstado() != null) {
            orden.setEstado(request.getEstado());
        }
        orden.setFechaProgramada(request.getFechaProgramada());
        orden.setFechaInicio(request.getFechaInicio());
        orden.setFechaCompletada(request.getFechaCompletada());
        orden.setCostoEstimado(request.getCostoEstimado());
        orden.setCostoFinal(request.getCostoFinal());
        orden.setNotasTecnicas(request.getNotasTecnicas());
        orden.setNotasCierre(request.getNotasCierre());

        // Registrar cambio de estado si aplica
        if (request.getEstado() != null && !request.getEstado().equals(estadoAnterior)) {
            registrarSeguimiento(orden.getId(), estadoAnterior, request.getEstado(), "Estado actualizado");
        }

        orden = ordenRepository.save(orden);
        log.info("Orden de mantenimiento actualizada: {}", orden.getId());
        return OrdenMantenimientoDTO.fromEntity(orden);
    }

    @Transactional
    public OrdenMantenimientoDTO cambiarEstadoOrden(Long id, CambiarEstadoRequest request) {
        Long empresaId = TenantContext.getCurrentTenant();
        OrdenMantenimiento orden = ordenRepository.findById(id)
                .filter(o -> o.getEmpresaId().equals(empresaId))
                .orElseThrow(() -> new EntityNotFoundException("Orden de mantenimiento no encontrada"));

        EstadoOrden estadoAnterior = orden.getEstado();
        orden.setEstado(request.getEstado());

        // Actualizar fechas según el estado
        if (request.getEstado() == EstadoOrden.EN_PROCESO && orden.getFechaInicio() == null) {
            orden.setFechaInicio(LocalDate.now());
        } else if (request.getEstado() == EstadoOrden.COMPLETADA) {
            orden.setFechaCompletada(LocalDate.now());
            if (request.getCostoFinal() != null) {
                orden.setCostoFinal(request.getCostoFinal());
            }
            if (request.getNotasCierre() != null) {
                orden.setNotasCierre(request.getNotasCierre());
            }
        }

        // Registrar seguimiento
        registrarSeguimiento(orden.getId(), estadoAnterior, request.getEstado(), request.getComentario());

        orden = ordenRepository.save(orden);
        log.info("Estado de orden {} cambiado de {} a {}", orden.getId(), estadoAnterior, request.getEstado());
        return OrdenMantenimientoDTO.fromEntity(orden);
    }

    @Transactional
    public void deleteOrden(Long id) {
        Long empresaId = TenantContext.getCurrentTenant();
        OrdenMantenimiento orden = ordenRepository.findById(id)
                .filter(o -> o.getEmpresaId().equals(empresaId))
                .orElseThrow(() -> new EntityNotFoundException("Orden de mantenimiento no encontrada"));

        seguimientoRepository.deleteByOrdenId(id);
        ordenRepository.delete(orden);
        log.info("Orden de mantenimiento eliminada: {}", id);
    }

    // ==================== SEGUIMIENTO ====================

    @Transactional(readOnly = true)
    public List<SeguimientoOrdenDTO> findSeguimientoByOrden(Long ordenId) {
        return seguimientoRepository.findByOrdenIdOrderByFechaRegistroDesc(ordenId)
                .stream()
                .map(SeguimientoOrdenDTO::fromEntity)
                .collect(Collectors.toList());
    }

    private void registrarSeguimiento(Long ordenId, EstadoOrden estadoAnterior, EstadoOrden estadoNuevo, String comentario) {
        SeguimientoOrden seguimiento = SeguimientoOrden.builder()
                .ordenId(ordenId)
                .estadoAnterior(estadoAnterior)
                .estadoNuevo(estadoNuevo)
                .comentario(comentario)
                .build();
        seguimientoRepository.save(seguimiento);
    }

    // ==================== ESTADÍSTICAS ====================

    @Transactional(readOnly = true)
    public Map<String, Object> getEstadisticas() {
        Long empresaId = TenantContext.getCurrentTenant();
        Map<String, Object> stats = new HashMap<>();

        stats.put("proveedoresActivos", proveedorRepository.countActivosByEmpresaId(empresaId));
        stats.put("ordenesPendientes", ordenRepository.countByEmpresaIdAndEstado(empresaId, EstadoOrden.PENDIENTE));
        stats.put("ordenesEnProceso", ordenRepository.countByEmpresaIdAndEstado(empresaId, EstadoOrden.EN_PROCESO));
        stats.put("ordenesCompletadas", ordenRepository.countByEmpresaIdAndEstado(empresaId, EstadoOrden.COMPLETADA));

        // Costos del mes actual
        LocalDate inicioMes = LocalDate.now().withDayOfMonth(1);
        LocalDate finMes = inicioMes.plusMonths(1).minusDays(1);
        BigDecimal costosMes = ordenRepository.sumCostosCompletadosEnRango(empresaId, inicioMes, finMes);
        stats.put("costosMesActual", costosMes != null ? costosMes : BigDecimal.ZERO);

        return stats;
    }
}

File: frontend/src/services/mantenimientoService.ts
import api from './api';

export type CategoriaMantenimiento =
  | 'PLOMERIA'
  | 'ELECTRICIDAD'
  | 'CARPINTERIA'
  | 'PINTURA'
  | 'LIMPIEZA'
  | 'JARDINERIA'
  | 'CERRAJERIA'
  | 'AIRE_ACONDICIONADO'
  | 'OTROS';

export type PrioridadOrden = 'BAJA' | 'MEDIA' | 'ALTA' | 'URGENTE';

export type EstadoOrden = 'PENDIENTE' | 'EN_PROCESO' | 'COMPLETADA' | 'CANCELADA';

export interface Proveedor {
  id: number;
  empresaId: number;
  nombre: string;
  razonSocial?: string;
  rfc?: string;
  telefonoPrincipal?: string;
  telefonoSecundario?: string;
  email?: string;
  direccion?: string;
  codigoPostal?: string;
  ciudad?: string;
  estado?: string;
  nombreContacto?: string;
  categorias: CategoriaMantenimiento[];
  notas?: string;
  activo: boolean;
  fechaCreacion: string;
  fechaActualizacion?: string;
}

export interface CreateProveedorRequest {
  nombre: string;
  razonSocial?: string;
  rfc?: string;
  telefonoPrincipal?: string;
  telefonoSecundario?: string;
  email?: string;
  direccion?: string;
  codigoPostal?: string;
  ciudad?: string;
  estado?: string;
  nombreContacto?: string;
  categorias?: CategoriaMantenimiento[];
  notas?: string;
}

export interface UpdateProveedorRequest extends CreateProveedorRequest {
  activo?: boolean;
}

export interface OrdenMantenimiento {
  id: number;
  empresaId: number;
  numeroOrden: string;
  propiedadId: number;
  propiedadNombre?: string;
  proveedorId?: number;
  proveedorNombre?: string;
  solicitanteId?: number;
  solicitanteNombre?: string;
  titulo: string;
  descripcion: string;
  categoria: CategoriaMantenimiento;
  prioridad: PrioridadOrden;
  estado: EstadoOrden;
  fechaSolicitud: string;
  fechaProgramada?: string;
  fechaInicio?: string;
  fechaCompletada?: string;
  costoEstimado?: number;
  costoFinal?: number;
  notasTecnicas?: string;
  notasCierre?: string;
  fechaCreacion: string;
  fechaActualizacion?: string;
  creadoPor?: string;
}

export interface CreateOrdenRequest {
  propiedadId: number;
  proveedorId?: number;
  solicitanteId?: number;
  titulo: string;
  descripcion: string;
  categoria: CategoriaMantenimiento;
  prioridad: PrioridadOrden;
  fechaProgramada?: string;
  costoEstimado?: number;
  notasTecnicas?: string;
}

export interface UpdateOrdenRequest {
  proveedorId?: number;
  solicitanteId?: number;
  titulo: string;
  descripcion: string;
  categoria: CategoriaMantenimiento;
  prioridad: PrioridadOrden;
  estado?: EstadoOrden;
  fechaProgramada?: string;
  fechaInicio?: string;
  fechaCompletada?: string;
  costoEstimado?: number;
  costoFinal?: number;
  notasTecnicas?: string;
  notasCierre?: string;
}

export interface CambiarEstadoRequest {
  estado: EstadoOrden;
  comentario?: string;
  costoFinal?: number;
  notasCierre?: string;
}

export interface SeguimientoOrden {
  id: number;
  ordenId: number;
  estadoAnterior?: EstadoOrden;
  estadoNuevo: EstadoOrden;
  comentario?: string;
  usuario?: string;
  fechaRegistro: string;
}

export interface EstadisticasMantenimiento {
  proveedoresActivos: number;
  ordenesPendientes: number;
  ordenesEnProceso: number;
  ordenesCompletadas: number;
  costosMesActual: number;
}

export const mantenimientoService = {
  // ==================== PROVEEDORES ====================

  getAllProveedores: async (): Promise<Proveedor[]> => {
    const response = await api.get('/mantenimiento/proveedores');
    return response.data;
  },

  getProveedoresActivos: async (): Promise<Proveedor[]> => {
    const response = await api.get('/mantenimiento/proveedores/activos');
    return response.data;
  },

  getProveedorById: async (id: number): Promise<Proveedor> => {
    const response = await api.get(`/mantenimiento/proveedores/${id}`);
    return response.data;
  },

  getProveedoresByCategoria: async (categoria: CategoriaMantenimiento): Promise<Proveedor[]> => {
    const response = await api.get(`/mantenimiento/proveedores/categoria/${categoria}`);
    return response.data;
  },

  createProveedor: async (data: CreateProveedorRequest): Promise<Proveedor> => {
    const response = await api.post('/mantenimiento/proveedores', data);
    return response.data;
  },

  updateProveedor: async (id: number, data: UpdateProveedorRequest): Promise<Proveedor> => {
    const response = await api.put(`/mantenimiento/proveedores/${id}`, data);
    return response.data;
  },

  deleteProveedor: async (id: number): Promise<void> => {
    await api.delete(`/mantenimiento/proveedores/${id}`);
  },

  // ==================== ÓRDENES ====================

  getAllOrdenes: async (): Promise<OrdenMantenimiento[]> => {
    const response = await api.get('/mantenimiento/ordenes');
    return response.data;
  },

  getOrdenesActivas: async (): Promise<OrdenMantenimiento[]> => {
    const response = await api.get('/mantenimiento/ordenes/activas');
    return response.data;
  },

  getOrdenById: async (id: number): Promise<OrdenMantenimiento> => {
    const response = await api.get(`/mantenimiento/ordenes/${id}`);
    return response.data;
  },

  getOrdenesByPropiedad: async (propiedadId: number): Promise<OrdenMantenimiento[]> => {
    const response = await api.get(`/mantenimiento/ordenes/propiedad/${propiedadId}`);
    return response.data;
  },

  getOrdenesByProveedor: async (proveedorId: number): Promise<OrdenMantenimiento[]> => {
    const response = await api.get(`/mantenimiento/ordenes/proveedor/${proveedorId}`);
    return response.data;
  },

  getOrdenesByEstado: async (estado: EstadoOrden): Promise<OrdenMantenimiento[]> => {
    const response = await api.get(`/mantenimiento/ordenes/estado/${estado}`);
    return response.data;
  },

  getOrdenesProgramadas: async (inicio: string, fin: string): Promise<OrdenMantenimiento[]> => {
    const response = await api.get(`/mantenimiento/ordenes/programadas?inicio=${inicio}&fin=${fin}`);
    return response.data;
  },

  createOrden: async (data: CreateOrdenRequest): Promise<OrdenMantenimiento> => {
    const response = await api.post('/mantenimiento/ordenes', data);
    return response.data;
  },

  updateOrden: async (id: number, data: UpdateOrdenRequest): Promise<OrdenMantenimiento> => {
    const response = await api.put(`/mantenimiento/ordenes/${id}`, data);
    return response.data;
  },

  cambiarEstadoOrden: async (id: number, data: CambiarEstadoRequest): Promise<OrdenMantenimiento> => {
    const response = await api.patch(`/mantenimiento/ordenes/${id}/estado`, data);
    return response.data;
  },

  deleteOrden: async (id: number): Promise<void> => {
    await api.delete(`/mantenimiento/ordenes/${id}`);
  },

  // ==================== SEGUIMIENTO ====================

  getSeguimientoByOrden: async (ordenId: number): Promise<SeguimientoOrden[]> => {
    const response = await api.get(`/mantenimiento/ordenes/${ordenId}/seguimiento`);
    return response.data;
  },

  // ==================== ESTADÍSTICAS ====================

  getEstadisticas: async (): Promise<EstadisticasMantenimiento> => {
    const response = await api.get('/mantenimiento/estadisticas');
    return response.data;
  },

  // ==================== CATÁLOGOS ====================

  getCategorias: async (): Promise<CategoriaMantenimiento[]> => {
    const response = await api.get('/mantenimiento/categorias');
    return response.data;
  },

  getPrioridades: async (): Promise<PrioridadOrden[]> => {
    const response = await api.get('/mantenimiento/prioridades');
    return response.data;
  },

  getEstados: async (): Promise<EstadoOrden[]> => {
    const response = await api.get('/mantenimiento/estados');
    return response.data;
  },

  // ==================== HELPERS ====================

  getCategoriaLabel: (categoria: CategoriaMantenimiento): string => {
    const labels: Record<CategoriaMantenimiento, string> = {
      PLOMERIA: 'Plomeria',
      ELECTRICIDAD: 'Electricidad',
      CARPINTERIA: 'Carpinteria',
      PINTURA: 'Pintura',
      LIMPIEZA: 'Limpieza',
      JARDINERIA: 'Jardineria',
      CERRAJERIA: 'Cerrajeria',
      AIRE_ACONDICIONADO: 'Aire Acondicionado',
      OTROS: 'Otros',
    };
    return labels[categoria] || categoria;
  },

  getPrioridadLabel: (prioridad: PrioridadOrden): string => {
    const labels: Record<PrioridadOrden, string> = {
      BAJA: 'Baja',
      MEDIA: 'Media',
      ALTA: 'Alta',
      URGENTE: 'Urgente',
    };
    return labels[prioridad] || prioridad;
  },

  getEstadoLabel: (estado: EstadoOrden): string => {
    const labels: Record<EstadoOrden, string> = {
      PENDIENTE: 'Pendiente',
      EN_PROCESO: 'En Proceso',
      COMPLETADA: 'Completada',
      CANCELADA: 'Cancelada',
    };
    return labels[estado] || estado;
  },

  getPrioridadColor: (prioridad: PrioridadOrden): 'default' | 'info' | 'warning' | 'error' => {
    const colors: Record<PrioridadOrden, 'default' | 'info' | 'warning' | 'error'> = {
      BAJA: 'default',
      MEDIA: 'info',
      ALTA: 'warning',
      URGENTE: 'error',
    };
    return colors[prioridad] || 'default';
  },

  getEstadoColor: (estado: EstadoOrden): 'default' | 'primary' | 'success' | 'error' => {
    const colors: Record<EstadoOrden, 'default' | 'primary' | 'success' | 'error'> = {
      PENDIENTE: 'default',
      EN_PROCESO: 'primary',
      COMPLETADA: 'success',
      CANCELADA: 'error',
    };
    return colors[estado] || 'default';
  },
};

export default mantenimientoService;
